src/status_im/contexts/wallet/wallet_connect/processing_events.cljs

```diff
@@ -0,0 +1,76 @@
+            [taoensso.timbre :as log]
+            [utils.transforms :as transforms]))
+
+(def ^:private method-to-screen
+  {constants/wallet-connect-personal-sign-method     :screen/wallet-connect.sign-message
+   constants/wallet-connect-eth-sign-typed-method    :screen/wallet-connect.sign-message
+   constants/wallet-connect-eth-sign-method          :screen/wallet-connect.sign-message
+   constants/wallet-connect-eth-sign-typed-v4-method :screen/wallet-connect.sign-message})
+
+(rf/reg-event-fx
+ :wallet-connect/process-session-request
+ (fn [{:keys [db]} [event]]
+   (let [method (wallet-connect-core/get-request-method event)
+         screen (method-to-screen method)]
+     (if screen
+       {:db (assoc-in db [:wallet-connect/current-request :event] event)
+        :fx [(condp = method
+               constants/wallet-connect-personal-sign-method
+               [:dispatch [:wallet-connect/process-personal-sign]]
+
+               constants/wallet-connect-eth-sign-method
+               [:dispatch [:wallet-connect/process-eth-sign]]
+
+               constants/wallet-connect-eth-sign-typed-method
+               [:dispatch [:wallet-connect/process-sign-typed]]
+
+               constants/wallet-connect-eth-sign-typed-v4-method
+               [:dispatch [:wallet-connect/process-sign-typed]])
+             [:dispatch [:open-modal screen]]]}
+       (log/error "Didn't find screen for Wallet Connect method"
+                  {:method method
+                   :event  :wallet-connect/process-session-request})))))
+
+(rf/reg-event-fx
+ :wallet-connect/process-personal-sign
+ (fn [{:keys [db]}]
+   (let [[raw-data address] (wallet-connect-core/get-db-current-request-params db)
+         parsed-data        (native-module/hex-to-utf8 raw-data)]
+     {:db (update-in db
+                     [:wallet-connect/current-request]
+                     assoc
+                     :address      address
+                     :raw-data     raw-data
+                     :display-data (or parsed-data raw-data))})))
+
+(rf/reg-event-fx
+ :wallet-connect/process-eth-sign
+ (fn [{:keys [db]}]
+   (let [[address raw-data] (wallet-connect-core/get-db-current-request-params db)
+         parsed-data        (native-module/hex-to-utf8 raw-data)]
+     {:db (update-in db
+                     [:wallet-connect/current-request]
+                     assoc
+                     :address      address
+                     :raw-data     raw-data
+                     :display-data (or parsed-data raw-data))})))
+
+(rf/reg-event-fx
+ :wallet-connect/process-sign-typed
+ (fn [{:keys [db]}]
+   (let [[address raw-data] (wallet-connect-core/get-db-current-request-params db)
+         parsed-data        (-> raw-data
+                                transforms/json->clj
+                                (dissoc :types :primaryType)
+                                (transforms/clj->pretty-json 2))]
```

Comment:
Have you checked a way to avoid this double transformation just to pretty print to JSON? Why do we need to do the expensive transformation to clj data structures just to throw away later?

I was wondering if you could just use `js-delete` to get those two keys out of the js object and then pretty print directly without the intermediary step `json->clj`.

Reply:
yeah would be much better to avoid transforming it back and forth. Will fix 👍

Reply:
`js-delete` is mutable, so had to add a utility to remove keys from js objects without mutating the original